src/pudl/output/ferc1.py

```diff
@@ -228,6 +228,55 @@ def _out_ferc1__yearly_plants_utilities(
     )
 
 
+## example/draft factory pattern
+specs = [
+    {
+        "name": "out_ferc1__yearly_purchased_power_and_exchanges_sched326",
+        "df1": "core_ferc1__yearly_purchased_power_and_exchanges_sched326",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+    {
+        "name": "out_ferc1__yearly_plant_in_service_sched204",
+        "df1": "core_ferc1__yearly_plant_in_service_sched204",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+    {
+        "name": "out_ferc1__yearly_balance_sheet_assets_sched110",
+        "df1": "core_ferc1__yearly_balance_sheet_assets_sched110",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+]
+
+
+# draft asset_factory
+def generate_asset_factory(spec):
+    @asset(io_manager_key="pudl_io_manager", compute_kind="Python")
+    def _asset() -> pd.DataFrame:
+        """Pull a useful dataframe of FERC Form 1 Purchased Power data."""
+        return_df = (
+            spec["df1"].merge(spec["df2"], on=spec["mg"])  # .pipe(
+            #     pudl.helpers.organize_cols,
+            #     [
+            #         "report_year",
+            #         "utility_id_ferc1",
+            #         "utility_id_pudl",
+            #         "utility_name_ferc1",
+            #         "seller_name",
+            #         "record_id",
+            #     ],
+            # )
```

Comment:
Yes, the `organize_cols` can and should be removed from *all* of the asset definitions in this module, since the database schema is now what determines the ordering of the columns.

Reply:
@zaneselvans If all of the helpers.organize_cols can be removed, is it necessary to make an asset factory for this? It looks to me like all of the functions are essentially doing the same thing then; they take two DataFrames as inputs, merge them on something hard-coded and the same across all the functions, and then return.

Is there a reason why we can't we just simplify them all into one function and change where the functions are called?

Reply:
We do need an asset factory that pumps out the function definitions, because those `@asset` decorated function definitions and their inputs and outputs are what define the dependency graph in Dagster. It might help to read through the [core Dagster concepts](https://docs.dagster.io/concepts) or some of their examples, and also to try running the ETL through the Dagster UI locally if you haven't already.

Reply:
Of course each of the assets (functions) defined by the factory will be very similar (which is why we want to use a factory) but it's the additional information from the decorator or the arguments and return value that stitch the functions and the data flowing through them together.